### PR TITLE
Fix #6154 - Python Client to PATCH descriptions

### LINKED ISSUE #6154: Python Client to PATCH descriptions
add a flag to force adding a description if there is already one.

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

Fix #6154 

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/utils/helpers.py [hunk 1]
```diff
@@ -270,3 +271,12 @@ def find_in_list(element: Any, container: Iterable[Any]) -> Optional[Any]:
...
     return next(iter([elem for elem in container if elem == element]), None)
+
+
+def find_column_in_table(column_name: str, table: Table) -> Optional[Column]:
+    """
+    If the column exists in the table, return it
+    """
+    return next(
+        iter([col for col in table.columns if col.name.__root__ == column_name]), None
+    )
```
Comment (TeddyCr) [inline ingestion/src/metadata/utils/helpers.py:282]: For these expression you can simply use a generator expression

```
return next((col for col in table.columns if col.name.__root__ == column_name), None)
```

File: ingestion/src/metadata/ingestion/ometa/mixins/patch_mixin.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            force: if True, we will patch any existing description. Otherwise, we will maintain
+                the existing data.
+        Returns
+            Updated Entity
+        """
+        instance = self._validate_instance_description(
+            entity=entity, entity_id=entity_id, force=force
+        )
+        if not instance:
+            return
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/ometa/mixins/patch_mixin.py:109]: I think we should return `None` explicitly just as a visual cue that we have a different path and that we are not breaking out of a loop

File: ingestion/src/metadata/utils/helpers.py (hunk 1 above)
Comment (pmbrull) [reply ingestion/src/metadata/utils/helpers.py:282]: such a good point here. I wrote this line down a few months ago and never looked back, thanks for the improvement, it is great

**Review by TeddyCr — APPROVED** [review]
LGTM 🚀

**Pull request merged**